src/services/wallets/tweb3.ts

```diff
@@ -0,0 +1,82 @@
+import Web3 from "web3";
+import Torus from "@toruslabs/torus-embed";
+import { provider } from "web3-core";
+
+export type BuildEnvironment =
+  | "production"
+  | "development"
+  | "binance"
+  | "testing"
+  | "lrc"
+  | "beta"
+  | undefined;
+
+// interface TorusCtorArgs in torus-embeded types
+const torusSettings = {
+  //buttonPosition?: 'top-left' | 'top-right' | 'bottom-right' | 'bottom-left'
+  //modalZIndex?: number
+  //apiKey?: string
+};
+
+// interface WhiteLabelParams in torus-embeded types
+// example data structure here: https://github.com/torusresearch/torus-embed/blob/master/examples/vue-app/src/data.js
+const popupStyling = {
+  theme: {
+    isDark: false,
+    colors: {
+      torusBrand1: "#000000",
+      torusGray2: "#FBF7F3",
+    },
+  },
+  logoDark: "https://www.projectliberty.io/assets/images/brand-icon.png",
+  logoLight: "https://www.projectliberty.io/assets/images/brand-icon.png",
+  //topupHide?: boolean
+  //featuredBillboardHide?: boolean
+  //disclaimerHide?: boolean
+  //tncLink?: LocaleLinks<string>
+  //privacyPolicy?: LocaleLinks<string>
+  //contactLink?: LocaleLinks<string>
+  //customTranslations?: LocaleLinks<any>
+};
+
+// interface TorusParams in trous-embeded types
+const initSettings = (buildEnv: BuildEnvironment) => {
+  return {
+    buildEnv: buildEnv || "production",
+    network: {
+      host: "http://localhost:7545",
+      chainId: 1337,
+      networkName: "Localchain",
+    },
+    showTorusButton: true,
+    whiteLabel: popupStyling,
+  };
+};
+
+const web3Torus = {
+  web3: new Web3(),
+  torus: null as Torus | null,
+  initialized: false,
+  setweb3: function (provider: provider): void {
+    const web3Inst = new Web3(provider);
+    web3Torus.web3 = web3Inst;
+    this.web3.setProvider(provider);
+  },
+  initialize: async function (buildEnv: BuildEnvironment): Promise<void> {
+    const torus = new Torus(torusSettings);
+    await torus.init(initSettings(buildEnv));
+    this.initialized = true;
+    try {
+      await torus.login({ verifier: undefined });
+      web3Torus.setweb3(torus.provider);
+      web3Torus.torus = torus;
+      sessionStorage.setItem("pageUsingTorus", buildEnv || "");
```

Comment:
maybe we can move this out of here so we can isolate the responsibility of this method.

Reply:
That's the plan moving forward in the sessionStorage and wallet API stories. The current wallet in use will be stored in the user's data which will be kept in sessionStorage.

For this story though, I didn't want to mess with it since it's outside the scope.

Reply:
Sometimes it is necessary to merge things in fast, is this is one of those cases?

If we merge this in we are creating more code debt. This PR is a result of code debt and I think it's probably best we don't create more. 

It might be better if we make the update now since it would probably take more effort to write up a story and plan for it.

Let me know if you wanna pair.

😄

Reply:
It's not about getting this one in fast that code was there already. Actually this line has been around for awhile, It just looks new because the file was converted to typescript.

My thinking is that while this does need to be fixed so does session storage as a whole. That story has already been planned and created and is bigger than this small piece. Since that's already going to need to be done regardless, this will be fixed at that point. 

Now the code should work if I just delete the line but then I might forget to integrate tracking of Torus into session storage later. Relying on my ability to remember seems worse than leaving it this way till the official story to change it comes up.

Reply:
Oh I see. I think it is better to delete it then if its not needed. You can make a note on the ticket as a reminder.

Reply:
Yeah I can do that. I'll delete it here and add a comment in the story so I don't forget